### PR TITLE
Gitlab collector: Database wraparound collector and test

### PR DESCRIPTION
**Pull request by Sticksman**
Split from #819

### CODE REVIEW

File: collector/pg_database_wraparound.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+var (
+	databaseWraparoundAgeDatfrozenxid = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, databaseWraparoundSubsystem, "age_datfrozenxid_seconds"),
+		"Age of the oldest transaction ID that has not been frozen.",
+		[]string{"datname"},
+		prometheus.Labels{},
+	)
+	databaseWraparoundAgeDatminmxid = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, databaseWraparoundSubsystem, "age_datminmxid"),
```
Comment (sysadmind) [inline collector/pg_database_wraparound.go:46]: Should this metric not also be `_seconds`?

File: collector/pg_database_wraparound.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		"Age of the oldest multi-transaction ID that has been replaced with a transaction ID.",
+		[]string{"datname"},
+		prometheus.Labels{},
+	)
+
+	databaseWraparoundQuery = `
+	SELECT
+		datname,
+		age(d.datfrozenxid) as age_datfrozenxid,
+		mxid_age(d.datminmxid) as age_datminmxid
```
Comment (sysadmind) [inline collector/pg_database_wraparound.go:56]:
```suggestion
		age(d.datfrozenxid) AS age_datfrozenxid,
		mxid_age(d.datminmxid) AS age_datminmxid
```

File: collector/pg_database_wraparound.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		databaseWraparoundQuery)
+
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var datname sql.NullString
+		var ageDatfrozenxid, ageDatminmxid sql.NullFloat64
```
Comment (sysadmind) [inline collector/pg_database_wraparound.go:76]: I think these values being NULL and still recording a metric is misleading.

Comment (Sticksman) [reply collector/pg_database_wraparound.go:76]: Perhaps we skip if datname is null?

**Review by SuperQ — APPROVED** [review]

**Review by sysadmind — APPROVED** [review]

**Pull request merged**